en/docs/cookies/global_tensor.md

```diff
@@ -0,0 +1,513 @@
+
+至此，本文从 Global Tensor 的创建开始，最终完成了一个基于 Global Tensor 的数据并行计算流程。
+
+So, this article will come to a close, and it fisrtly introduces how to create a global tensor and finally explains the detailed steps for data parallelism computation that is based on a global tensor.
+
+更多并行方式和 SBP 的推理逻辑，将在后续内容介绍。
+
+More about parallelism ways and SBP's inference logic will be discussed in our later articles. 
+
+## 扩展阅读
+
+## Further reading
+
+### OneFlow 多机多卡启动 和 依赖的环境变量
+
+### OneFlow’s multi-machine multi-GPU launching and its required environment variables
+
+OneFlow 的 Global Tensor 执行采用的是 **多客户端模式 (Multi-Client)**，每个设备对应一个进程。`n 机 m 卡` 的环境，就对应 `n * m` 个进程。每个进程都有一个进程 rank 编号，Global Tensor 中的 placement 参数中的 ranks 对应的就是这个 rank 编号。
+
+OneFlow’s global tensors are executed under ** Multi-Client mode**, which means each device corresponds to one process. For example, `n Machine m GPU` has `n * m` processes. Besides, each process has its own rank ID, which corresponds to the ranks of the global tensor's placement parameter.
+
+以 `2 机 2 卡` 为例， 0 号机器中两张卡分别对应编号 0 和 1，第 1 号机器中两张卡分别对应编号 2 和 3。此时 `flow.placement(type="cuda", ranks=[2])` 可以唯一标识第 1 号机器中的第 0 卡。
+
+Taking `2 Machines 2 GPUs` for example, Machine 0 corresponds to GPU 0 and GPU 1, and Machine 1 corresponds to GPU 2 and GPU 3. So, `flow.placement(type="cuda", ranks=[2])` can only identify the GPU 0 on Machine 1.
+
+一般地，对于 `n 机 m 卡` 的环境，`flow.placement(type="cuda", ranks=[k])` 唯一标识第 `k / n` 号机器的第 `k % m` 张卡。
+
+Generally, in the `n Machine m GPU` environment, `flow.placement(type="cuda", ranks=[k])` only identifies the GPU `k % m` on Machine `k / n`.
+
+因为采用多客户端模式，所以需要对应每个设备都启动一个进程。在 OneFlow 中，所有进程都只需要启动相同的脚本程序。不同进程之间通过不同的环境变量来区分进程编号和建立通信连接。
+
+Because the tensor adopts the Multi-Client mode, we need to launch different processes corresponding to each device. In OneFlow, all processes need to launch the same scripts, and different processes distinguish process ID and establish communications according to different environment variables.
+
+环境变量说明：
+
+Notes of environment variables:
+
+- `MASTER_ADDR`：多机训练的第 0 号机器的 IP；
+- `MASTER_PORT`：多机训练的第 0 号机器的监听端口，不与已经占用的端口冲突即可；
+- `WORLD_SIZE`：整个集群中计算设备的数目，因为目前还不支持各个机器上显卡数目不一致，因此 `WORLD_SIZE` 的数目实际上是 $机器数目 \times 每台机器上的显卡数目$。[创建 Global Tensor](#global-tensor_2) 中的示例是单机 2 卡的情况，因此 `WORLD_SIZE=2`；
+- `RANK`：集群内所有机器下的进程编号；
+- `LOCAL_RANK`：单个机器内的进程编号；
+
+- `MASTER_ADDR`：the IP of Machine 0 that is trained on multiple machines;
+- `MASTER_PORT`：the monitor port of Machine 0 that is trained on multiple machines, and this port shouldn’t conflict with the occupied ports;
+- `WORLD_SIZE`: the number of computing device in the whole cluster. Because it’s still not feasible to configure different number of GPUs on each device, the `WORLD_SIZE` equals the machine numbers multiplies the GPU numbers on each machine. In the previous case, we [create the global tensor](#global-tensor_2) in single machine 2 GPUs environment, so the `WORLD_SIZE=2`;
```

Comment:
```suggestion
- `WORLD_SIZE`: the number of computing devices in the whole cluster. Because it’s still not feasible to configure different number of GPUs on each device, the `WORLD_SIZE` equals the machine numbers multiplies the GPU numbers on each machine. In the previous case, we [create the global tensor](#global-tensor_2) in single machine 2 GPUs environment, so the `WORLD_SIZE=2`;
```